crates/rover-client/src/operations/graph/check/mutation_runner.rs

```diff
@@ -1,76 +1,65 @@
+use crate::operations::graph::check::types::{
+    GraphCheckInput, MutationChangeSeverity, MutationResponseData,
+};
+use crate::shared::CheckResponse;
 use crate::RoverClientError;
-use graphql_client::*;
 
-use reqwest::Url;
+use graphql_client::*;
 
 type Timestamp = String;
 #[derive(GraphQLQuery)]
 // The paths are relative to the directory where your `Cargo.toml` is located.
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
-    query_path = "src/query/graph/check.graphql",
+    query_path = "src/operations/graph/check/check_mutation.graphql",
     schema_path = ".schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
 /// This struct is used to generate the module containing `Variables` and
 /// `ResponseData` structs.
-/// Snake case of this name is the mod name. i.e. check_schema_query
-pub struct CheckSchemaQuery;
+/// Snake case of this name is the mod name. i.e. graph_check_mutation
+pub(crate) struct GraphCheckMutation;
 
 /// The main function to be used from this module.
 /// This function takes a proposed schema and validates it against a published
 /// schema.
 pub fn run(
-    variables: check_schema_query::Variables,
+    input: GraphCheckInput,
     client: &StudioClient,
 ) -> Result<CheckResponse, RoverClientError> {
-    let graph = variables.graph_id.clone();
-    let data = client.post::<CheckSchemaQuery>(variables)?;
+    let graph = input.graph_id.clone();
+    let data = client.post::<GraphCheckMutation>(input.into())?;
     get_check_response_from_data(data, graph)
 }
 
-#[derive(Debug)]
-pub struct CheckResponse {
-    pub target_url: Option<Url>,
-    pub number_of_checked_operations: i64,
-    pub change_severity: check_schema_query::ChangeSeverity,
-    pub changes: Vec<check_schema_query::CheckSchemaQueryServiceCheckSchemaDiffToPreviousChanges>,
-}
-
 fn get_check_response_from_data(
-    data: check_schema_query::ResponseData,
+    data: MutationResponseData,
     graph: String,
 ) -> Result<CheckResponse, RoverClientError> {
     let service = data.service.ok_or(RoverClientError::NoService { graph })?;
-    let target_url = get_url(service.check_schema.target_url);
+    let target_url = service.check_schema.target_url;
 
     let diff_to_previous = service.check_schema.diff_to_previous;
 
     let number_of_checked_operations = diff_to_previous.number_of_checked_operations.unwrap_or(0);
 
-    let change_severity = diff_to_previous.severity;
-    let changes = diff_to_previous.changes;
+    let change_severity = diff_to_previous.severity.into();
+    let mut changes = Vec::with_capacity(diff_to_previous.changes.len());
+    let mut num_failures = 0;
+    for change in diff_to_previous.changes {
+        if let MutationChangeSeverity::FAILURE = change.severity {
+            num_failures += 1;
+        }
+        changes.push(change.into());
+    }
 
     Ok(CheckResponse {
+        num_failures,
```

Comment:
The problem here is that we want to fail when there are failures, but we _also_ want to print out the table of changes that occurred. What I have here is kind of a stopgap where subgraph composition errors are returned as `RoverClientError`s, to promote proper usage of the library.

For regular client errors, those are just determined in `rover` right now, and it's very easy to just... ignore them since it's not returned as a proper error.

My plan is to have a new `RoverClientError::ClientCheckError { check_response: CheckResponse }` so that when failures are detected, they are returned as a proper error, but they will _also_ have all of the information that would be printed on a success. This should make it easy for `rover` to have something like:

```rust
match check_response {
  Ok(check_response) => {
    print_table(check_response);
    Ok(RoverStdout::None)
  },
  Err(e) => {
    print_table(e.check_response);
	anyhow!("{}", e");
  }
}
```